### PR TITLE
Fix bug in get datasource API and improve memory usage

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Fix bug in get datasource API and improve memory usage
 
### Issues Resolved
N/A
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by heemin32** [comment]
Build is failing due to https://github.com/opensearch-project/OpenSearch/pull/7508

File: src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacade.java [hunk 1]
```diff
@@ -368,14 +365,20 @@ public void putGeoIpData(
...
+        }
         while (iterator.hasNext()) {
             CSVRecord record = iterator.next();
-            String document = createDocument(fields, record.values());
-            IndexRequest request = Requests.indexRequest(indexName).source(document, XContentType.JSON);
-            bulkRequest.add(request);
+            XContentBuilder document = createDocument(fields, record.values());
+            IndexRequest indexRequest = (IndexRequest) requests.poll();
+            indexRequest.source(document);
+            indexRequest.id(record.get(0));
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacade.java:380]: is this get by index safe, or we should check for size first?

File: src/test/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceTransportActionTests.java [hunk 2]
```diff
@@ -5,18 +5,22 @@
...
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
+import javax.swing.*;
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceTransportActionTests.java:17]: is this import correct? I think we're not using *, plus swing package look suspicious?

File: src/test/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacadeTests.java [hunk 3]
```diff
@@ -52,6 +52,7 @@
 import org.opensearch.action.search.SearchResponse;
 import org.opensearch.action.support.IndicesOptions;
 import org.opensearch.common.Randomness;
+import org.opensearch.common.Strings;
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacadeTests.java:55]: please make sure if works with latest update from core, they start moving classes around https://github.com/opensearch-project/OpenSearch/pull/7508

File: src/test/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceTransportActionTests.java (hunk 2 above)
Comment (heemin32) [reply src/test/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceTransportActionTests.java:17]: Yes. Thanks for through review. Let me remove it.

File: src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacade.java [hunk 4]
```diff
@@ -204,31 +209,23 @@ protected CSVParser internalGetDatabaseReader(final DatasourceManifest manifest,
...
         }
-        StringBuilder sb = new StringBuilder();
-        sb.append("{\"");
-        sb.append(IP_RANGE_FIELD_NAME);
-        sb.append("\":\"");
-        sb.append(values[0]);
-        sb.append("\",\"");
-        sb.append(DATA_FIELD_NAME);
-        sb.append("\":{");
+        XContentBuilder builder = XContentFactory.jsonBuilder();
```
[the remark this answers is not in the document]
Comment (heemin32) [reply src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacade.java:218]: ack

File: src/test/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacadeTests.java (hunk 3 above)
Comment (heemin32) [reply src/test/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataFacadeTests.java:55]: Yes. There is another PR to fix it.https://github.com/opensearch-project/geospatial/pull/314

**Review by martin-gaievski — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**